test/duplicate_test.go

```
@@ -38,7 +38,7 @@ func TestDuplicatePodTaskRun(t *testing.T) {
 	defer tearDown(t, c, namespace)
 
 	var wg sync.WaitGroup
-	for i := 0; i < 25; i++ {
+	for i := 0; i < 5; i++ {
```

Reducing this number reduces the test's effectiveness -- now it needs duplicate pods to be created 1 in every 5 times to be able to detect it, as opposed to just 1 in every 25.

If the real underlying problem is OOMs, can we solve this by setting resource requests/limits on the TaskRuns, so they don't get OOM-killed?

---

I wonder if there is a better way to tests this now. Also, this was introduced when Tekton was backed by knative/build, which is not the case anymore..

We may want to try setting resources limit to see if it fixes things :angel:

---

maybe i don't understand how resources limits works, but isn't that if we don't specify any limit by default it will starve memory for the pod until the node's limits is getting reach?

---

@chmouel right, but it should also take though into account while scheduling, not allowing pod to be scheduled on a node that is starving already :angel:

---

am I understanding right that  either way we will should be still failing ? just failing early ?

---

well, it would wait for memory to be back to schedule new one if all nodes are starving, reducing the number of pods running in parallel at the same time :sweat_smile: … which may reduce the effectivness of the test too :sweat: :thinking:

---

Are we really sure it detects duplication tho ? It does sounds a lot of hoopla hoops to go thru to make sure k8 behave rights (ie: to have no races between task creation and start)

---

@chmouel it did back in the day for sure (aka this test against the code non fixed would fail for the good reason)